android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsViewModel.kt

```diff
@@ -54,8 +66,58 @@ class SurveyQuestionsViewModel(
     }
 
     fun answerQuestion(questionAndAnswers: QuestionAndAnswers) {
-        Timber.d(
-            "${questionAndAnswers.questionIndex} - ${questionAndAnswers.answerInputs}"
-        )
+        val uiModels = _questionContentUiModels.value.toMutableList()
+        val questionIndex = questionAndAnswers.questionIndex
+        val changedUiModel = uiModels[questionIndex]
+        changedUiModel.displayType.input = questionAndAnswers.answerInputs
+        uiModels[questionIndex] = changedUiModel
+        _questionContentUiModels.value = uiModels
+
+        // Convert answers to SurveySubmission
+        _survey?.let {
+            val questionSubmissions = questionContentUiModels.value
+                .mapIndexed { questionIndex, uiModel ->
+                    val questionId = questionId(questionIndex)
+                    val answerSubmissions: List<AnswerSubmission> =
+                        uiModel.displayType.input.map { input ->
+                            when (input) {
+                                is AnswerInput.Select -> AnswerSubmission(id = input.id)
+                                is AnswerInput.Content -> AnswerSubmission(
+                                    id = input.id,
+                                    answer = input.content
+                                )
+                            }
+                        }
+                    QuestionSubmission(
+                        questionId,
+                        answerSubmissions
+                    )
+                }
+            surveySubmission = SurveySubmission(it.id, questionSubmissions)
+        }
+    }
+
+    fun submitSurveyResponse() {
+        surveySubmission?.let { surveySubmission ->
+            val questionSubmissions = surveySubmission.questions.filter { it.answers.isNotEmpty() }
+            viewModelScope.launch {
+                _submittingAnswerState.value = SubmittingAnswerState.SUBMITTING
+                submitSurveyResponseUseCase(surveySubmission.copy(questions = questionSubmissions))
+                    .catch { e ->
+                        showError(e.message)
+                        _submittingAnswerState.value = SubmittingAnswerState.IDLE
+                    }
+                    .collect {
+                        _submittingAnswerState.value = SubmittingAnswerState.SUCCESS
+                    }
+            }
+        }
+    }
+
+    private fun questionId(index: Int): String {
+        return _survey?.sortedQuestions()
+            ?.elementAtOrNull(index)
+            ?.id
+            .orEmpty()
```

Comment:
Getting the question id by `index` is not safe if there is something wrong with the order, at least from my feeling. Should we find another way to generate and pass it here instead?

One of the better ways I can think of right now is adding a new field `id` into `QuestionContentUiModel`, and passing it there when we initialize QuestionContentUiModel from `Question` 💭 

```
data class QuestionContentUiModel(
    val progress: String,
    val title: String,
    val id: String,
    val displayType: QuestionDisplayType
)
```

<img width="396" alt="image" src="https://user-images.githubusercontent.com/16315358/210483592-dff73f20-1a75-4408-8d3a-88ae45451594.png">

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/123/commits/f972ffa3a07361be38a62957334b6171e77d267d